api/src/modules/button/button.service.ts

```diff
@@ -572,6 +572,22 @@ export class ButtonService {
     return this.buttonRepository.update(buttonId, { expired: true });
   }
 
+  /* setPin(buttonId: string, status: boolean) {
+    return this.buttonRepository.update(buttonId, { pin: status });
+  } */
+
+  async findByPin() {
+    let buttons: Button[] = await this.buttonRepository.find({
+      where: {
+        pin: true,
+        deleted: false,
+        expired: false,
+      },
```

Comment:
https://github.com/helpbuttons/helpbuttons/blob/main/api/src/modules/button/button.service.ts#L182

use the function expiredBlockedConditions cause it excludes buttons of blocked accounts